tests/actions/exitUnspent.js

```diff
@@ -10,145 +10,151 @@ const waitForBalanceChange = require('./waitForBalanceChange');
-        debug(youngestInputProof);
-    } else {
-        debug("No youngest input found. Will try to exit deposit");
-        youngestInputProof = [];
-    }
-    debug("------Period from the contract by merkle root------");
-    debug(await contracts.bridge.periods(proof[0]));
-    log("------Balance before exit------");
-    const balanceBefore = await contracts.token.balanceOf(addr);
-    const plasmaBalanceBefore = await node.getBalance(addr);
-    log("Account mainnet balance: ", balanceBefore);
-    log("Account plasma balance: ", plasmaBalanceBefore);
-    log("Attempting exit...");
-    let startExitResult =
-      await contracts.exitHandler.connect(wallet.provider.getSigner(addr)).startExit(
-        youngestInputProof,
-        proof,
-        unspent.outpoint.index,
-        youngestInput.index,
-        { value: ethers.utils.parseEther('1'), gasLimit: 2000000 }
-    );
-    await startExitResult.wait();
+  log(`------Unspents of ${addr}------`);
+  const unspents = await node.getUnspent(addr);
+  log(unspents);
+  debug("------Looking for unspent from submitted period------");
+  const latestBlockNumber = (await node.getBlock('latest')).number;
+  debug("Latest Block number: ", latestBlockNumber);
+  const latestSubmittedBlock = latestBlockNumber - latestBlockNumber % 32;
+  debug("Latest submitted block number: ", latestSubmittedBlock);
+  if (latestSubmittedBlock === 0) {
+      throw new Error("Can't exit, no periods were submitted yet");
+  };
+  
+  const getIndex = async (unspents, lastBlock) =>{
+      for(let i=0; i<unspents.length; i++) {
+          txHash = unspents[i].outpoint.hash;
+          txData = await node.getTransaction(bufferToHex(txHash));
+          debug("Unspent", i, "blocknumber:", txData.blockNumber);
+          debug("Is submitted?", txData.blockNumber < lastBlock);
+          if (txData.blockNumber < lastBlock) return i;
+      }
+  
+      return -1;
+  };
+
+  const unspentIndex = uIndex === undefined || uIndex === null ? await getIndex(unspents, latestSubmittedBlock) : uIndex;
+  console.log(unspentIndex);
+  if (unspentIndex === -1) {
+      throw new Error("Can't exit, no unspents are in submitted periods found");
+  };
+  log(`------Will attept to exit unspent ${unspentIndex} of ${addr}------`);
+  const unspent = unspents[unspentIndex];
+  txHash = unspent.outpoint.hash;
+  txData = await node.getTransaction(bufferToHex(txHash));
+  const amount = unspent.output.value;
+  const txColor = unspent.output.color;
+  log("Unspent amount: ", amount);
+  debug(`------Transaction hash for Bob's unspent ${unspentIndex}------`);
+  debug(txHash);
+  debug("------Transaction data------");
+  debug(txData);
+  debug("------Period------");
+  const period = await Period.periodForTx(node, txData);
+  debug(period);
+  debug("------Proof------");
+  const periodData = await plasmaWallet.provider.send('plasma_getPeriodByBlockHeight', [txData.blockNumber]);
```

Comment:
this is the essential change here — using period data from the node instead of hardcoded one. Sorry for the mess with whitespace

Reply:
reverted all the whitespace mess